### ISSUE TITLE
dict args should accept named tuples or at least provide a way to build a PyObject dict

### ISSUE THREAD

**Issue by timotheecour** [yglukhov/nimpy#216]
@yglukhov 
## example
I'm trying to pass arguments to `generate_presigned_url` which requires dict argument, see https://boto3.amazonaws.com/v1/documentation/api/latest/guide/s3-presigned-urls.html
```py
# python code
s3.generate_presigned_url('get_object', Params={'Bucket': bucket_name, 'Key': object_name})
```
using a tuple doesn't work:
```nim
# nim code
s3.generate_presigned_url("get_object", Params=(Bucket: bucket_name, Key: object_name))
```

IMO we should allow passing named tuples.

Secondly, is there a way to pass a `PyObject` directly, and construct it as a dict (eg in case keys can't be mapped to nim identifiers, or other reasons); eg:
```nim
# nim code
s3.generate_presigned_url("get_object", Params=dict({"Bucket": bucket_name, "Key": object_name}))
```
this doesn't work but how could we make this work?

Note: I'm not interested in a solution involving JsonNode, which isn't always possible and involves overhead.

## links
* related to https://github.com/yglukhov/nimpy/issues/133
unlike what was said in https://github.com/yglukhov/nimpy/issues/133#issuecomment-590050018, we can restrict named tuples via `typetraits.isNamedTuple` if needed; also

> Python dicts are better mapped to JsonNode if they are json-compatible

that's not always the case and involves un-necessary marshalling

> or objects if they always provide the fields

the problem is this requires creating a dummy object type just to pass arguments to an API expecting a dict

> or just PyObject to skip marshalling and work with the underlying object directly.

how would that work? example code would help

* https://github.com/yglukhov/nimpy/issues/66

**Comment by yglukhov** [comment]
I believe dict constructor should just work, it accepts kwargs, doesn't it? So no intermediate tuples are needed.

**Comment by timotheecour** [comment]
> I believe dict constructor should just work, it accepts kwargs, doesn't it? So no intermediate tuples are needed.

what would the code look like?

**Comment by yglukhov** [comment]
Smth like
```nim
let py = pyBuiltinsModule()
s3.generate_presigned_url("get_object", Params=py.dict(Bucket = bucket_name, Key = object_name))
```

**Comment by timotheecour** [comment]
thanks this would work, see also https://github.com/yglukhov/nimpy/pull/218

**Comment by yglukhov** [comment]
Fixed by introducing `toDict` in #218
